pkgs/build-support/docker/examples.nix

```diff
@@ -123,7 +123,14 @@ rec {
     finalImageTag = "2.2.1";
     finalImageName = "nix";
   };
-
+  # Same example again, but this time specify an os, an arch and a cpu variant
+  busyboxFromDockerHub = pkgsCross.armv7l-hf-multiplatform.dockerTools.pullImage {
+    imageName = "docker.io/library/busybox";
+    imageDigest = "sha256:ef320ff10026a50cf5f0213d35537ce0041ac1d96e9b7800bafd8bc9eff6c693";
+    sha256 = "sha256-9T6tP/xnZwaTH+qkbkVuXhCk+qZGE3WBY8T+Ug4qCmc=";
+    finalImageName = "docker.io/library/busybox";
+    finalImageTag = "stable-uclibc";
+  };
```

Comment:
There's `nixos/tests/docker-tools-cross.nix`, which uses a remote builder to cross compile to the system where the test runs. It's a bit mind bending, because the construction of the test starts with the host system rather than the build system, which is a bit unintuitive, but it works. Sadly, ofborg can't run it, because you need an appropriate remote builder to run it yourself. The ofborg builders aren't configured that way. Hydra does run the test and it works. https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.docker-tools-cross.x86_64-linux

Reply:
@utdemir Sorry for being late, I'm a little bit busy these times. I've updated the doc according to your request (I've not yet committed my changes), but there is one part I'm not sure to fully understand in your problem.

> So, when I am on a x86_64 machine, but want to produce Docker images that work on armv7, this is not what I want to use. Because if I run pkgsCross.armv7l-hf-multiplatform.dockerTools.streamLayeredImage, the streamLayeredImage executable that will emit the Docker image won't work on my computer

Despite what you claimed, I'm able to build and run the executable emitted by `streamLayeredImage` on my x86_64 computer, using `pkgs.pkgsCross.aarch64-multiplatform` to build and fetch the executable / docker image and then copy and successfully run the produced image on my PI4. Since the line you commented (down below) don't restrict the platforms it should be runs on (at least if I'm right). They are just being used to create the manifest config file for the image.

https://github.com/NixOS/nixpkgs/blob/84cf42812e82cc2d8879a45bfba9e62a1f74dc00/pkgs/build-support/docker/stream_layered_image.py#L354-L358

<details>
<summary>
Custom file used for testing
</summary>
<p>

```nix
{ pkgs, ... }:

pkgs.pkgsCross.aarch64-multiplatform.dockerTools.streamLayeredImage {
  name = "test";
  tag = "0.0.1";
  contents = [ pkgs.pkgsCross.aarch64-multiplatform.hello ];

  created = "now";

  fromImage = pkgs.pkgsCross.aarch64-multiplatform.dockerTools.pullImage {
    imageName = "docker.io/library/busybox";
    imageDigest = "sha256:ef320ff10026a50cf5f0213d35537ce0041ac1d96e9b7800bafd8bc9eff6c693";
    sha256 = "sha256-BjGm4uS98GhigVhtgEMJ4qMayqek/+cMSrCz3F2J4UQ=";
    finalImageName = "docker.io/library/busybox";
    finalImageTag = "stable-uclibc";
  };

  config = {
    WorkingDir = "/";

    Entrypoint = [ "/bin/hello" ];
  };
}
```

</p>
</details>